common/model/bundle.c

```diff
@@ -87,6 +140,81 @@ void bundle_finalize(bundle_transactions_t *bundle, Kerl *const kerl) {
   }
 }
 
+retcode_t bundle_validator(bundle_transactions_t *const bundle,
+                           bundle_status_t *const status) {
+  retcode_t res = RC_OK;
+  iota_transaction_t *curr_tx = NULL;
+  int64_t index = 0, last_index = 0;
+  int64_t bundle_value = 0;
+  flex_trit_t bundle_hash[FLEX_TRIT_SIZE_243];
+  bool valid_sig = false;
+  Kerl kerl = {};
+
+  if (bundle == NULL) {
+    *status = BUNDLE_NOT_INITIALIZED;
+    return RC_NULL_PARAM;
+  }
+
+  curr_tx = (iota_transaction_t *)utarray_eltptr(bundle, 0);
+  last_index = transaction_last_index(curr_tx);
+
+  if (utarray_len(bundle) != last_index + 1) {
+    *status = BUNDLE_INCOMPLETE;
+    return res;
+  }
+
+  memcpy(bundle_hash, transaction_bundle(curr_tx), FLEX_TRIT_SIZE_243);
+
+  BUNDLE_FOREACH(bundle, curr_tx) {
+    bundle_value += transaction_value(curr_tx);
+
+    if (llabs(bundle_value) > MAX_IOTA_SUPPLY) {
```

Comment:
we probably want to do this chack for the `transaction_value` as well, not just the bundle value which is an aggregate?

Reply:
I am confused why this buggy implementation was approved. turn out I need to fix it because I moved it.